plugins/dell/fu-dell-smi.c

```diff
@@ -7,8 +7,8 @@
 
 #include "config.h"
 
-#include <appstream-glib.h>
 #include "fu-dell-smi.h"
+#include <appstream-glib.h>
```

Comment:
Is there a rule for including local headers first? I always included system then local, but I'm not terribly sure of why.

Reply:
Just an options to `SortIncludes`, I don't see anything for system versus local.  If you feel strongly on this, we'll probably need to do a post process cleanup.

Reply:
I think I found what you're looking for and added it to clang format.

'Regroup'
```
IncludeBlocks (IncludeBlocksStyle)
Dependent on the value, multiple #include blocks can be sorted as one and divided based on category.

Possible values:

IBS_Preserve (in configuration: Preserve) Sort each #include block separately.

#include "b.h"               into      #include "b.h"

#include <lib/main.h>                  #include "a.h"
#include "a.h"                         #include <lib/main.h>
IBS_Merge (in configuration: Merge) Merge multiple #include blocks together and sort as one.

#include "b.h"               into      #include "a.h"
                                       #include "b.h"
#include <lib/main.h>                  #include <lib/main.h>
#include "a.h"
IBS_Regroup (in configuration: Regroup) Merge multiple #include blocks together and sort as one. Then split into groups based on category priority. See IncludeCategories.

#include "b.h"               into      #include "a.h"
                                       #include "b.h"
#include <lib/main.h>
#include "a.h"                         #include <lib/main.h>
```

Reply:
@hughsie 

> I always included system then local, but I'm not terribly sure of why.

I think this is a common guideline in many projects and I think the main reason is to prevent weird issues with user-defined macros affecting the system libs.

Reply:
It looks like that's what the default style will cause to happen with no regex for category ordering is put in as well.  Seems reasonable to me.